plugins/content/document/lib/operation_engines/KOperationEnginePdfCreator.php

```diff
@@ -96,19 +96,18 @@ public function operate(kOperator $operator = null, $inFilePath, $configFilePath
 		
 		$finalOutputPath = $this->outFilePath;
 		
-		if (($inputExtension == 'pdf') && ($this->data->flavorParamsOutput->readonly == true)){
-			$tmpFile = $this->outFilePath.'.pdf';
-		}else{
-			$tmpFile = kFile::replaceExt(basename($realInFilePath), 'pdf');
-			$tmpFile = dirname($this->outFilePath).'/'.$tmpFile;
-		}
+		$tmpFile = kFile::replaceExt(basename($realInFilePath), 'pdf');
+		$tmpFile = dirname($this->outFilePath).'/'.$tmpFile;
+
 		$this->outFilePath = $tmpFile;
 		
 		// Create popups log file
 		$killPopupsPath = $this->getKillPopupsPath();
-		if(file_exists($killPopupsPath))
+		if(!is_null($killPopupsPath) && file_exists($killPopupsPath))
+		{
 			unlink($killPopupsPath);
-		
+		}
+
 		// Test file type 
 		$errorMsg = $this->checkFileType($realInFilePath, $this->SUPPORTED_FILE_TYPES);
```

Comment:
Since your new worker may encounter un supported file types, in case that error is returned and file type is not supported, job must be stopped with Error, so we can monitor it and understand that we need to fix it.